RI-904: Mise a jour a expo 53

kaaloo opened the pull request:
## 🚀 Upgrade to Expo SDK 53 and React 19

### Summary
Upgraded the entire monorepo to Expo SDK 53 (mobile app) and React 19 (all packages), including all necessary dependency updates and React 19 compatibility fixes.

### Changes

#### Mobile App (Expo SDK 53)
- Upgraded Expo SDK from 52 to 53
- Upgraded React Native from 0.77.0 to 0.79.6
- Upgraded React from 18.3.1 to 19.0.0
- Upgraded Firebase packages to v23.4.0 (SDK 53 compatible)
- Updated all Expo packages via `expo install --fix`
- Regenerated iOS/Android native directories with new templates

#### Web Apps (React 19)
- Upgraded React to 19.0.0 in client, storybook, and UI package
- Updated `@types/react` to 19.0.14 and `@types/react-dom` to 19.2.1

#### TypeScript Configuration
- Updated mobile tsconfig to use `moduleResolution: "bundler"`
- Added `skipLibCheck: true` for compatibility
- Added pnpm overrides to deduplicate React types across monorepo

#### React 19 Compatibility Fixes
- **useRef**: Added required initial values (e.g., `useRef<T>(null)` → `useRef<T | null>(null)`)
- **Ref types**: Updated all ref-accepting functions/components to accept nullable refs (`RefObject<T | null>`)
- **React.cloneElement**: Added type assertions for stricter type checking
- **Expo notifications**: Updated `NotificationBehavior` with new required fields (`shouldShowBanner`, `shouldShowList`)
- **Theme types**: Fixed `theme.isRTL` → `theme.i18n.isRTL` access pattern

### Testing
- ✅ All TypeScript type checks passing
- ✅ All ESLint checks passing
- ✅ Mobile app prebuild successful
- ✅ Client and storybook dev servers running

### Breaking Changes
None - all changes are backward compatible within the codebase.

### Notes
- Some peer dependency warnings exist for third-party libraries expecting React 18, but these are non-blocking
- React 19's stricter typing improves type safety across the codebase
- New Architecture is now default in Expo SDK 53

kaaloo committed:
chore: upgrade dependencies including React 19, Expo 53, and Firebase 23.4

kaaloo committed:
refactor: replace GoogleAPISuggestion with strongly-typed GeoAPISuggestion interface

kaaloo committed:
fix: force unique react types to fix mobile check type errors

kaaloo committed:
fix(mobile): fix type checking

kaaloo committed:
fix: remaning check types issues

kaaloo committed:
chore: upgrade React and React DOM to v19.0.0

kaaloo committed:
test(client): :white_check_mark: Update snapshot

kaaloo committed:
chore: add @firebase/util to pnpm built dependencies list

kaaloo committed:
fix: enable package exports in metro config for redux-saga compatibility

kaaloo committed:
chore: disable new architecture in Expo mobile app config

kaaloo committed:
chore: simplify metro config by removing unused unstable_conditionNames flag

kaaloo committed:
fix: resolve redux-saga package exports compatibility in Metro bundler

kaaloo committed:
feat: enable new Expo architecture in mobile app config

ledjay commented on packages/ui/src/components/composites/annotations-overlay/AnnotationsOverlay.tsx:
```diff
@@ -17,7 +17,7 @@ export const AnnotationsOverlay = ({ children, className, annotations }: Annotat
   return (
     <figure className={cn("relative", className)}>
-      {React.cloneElement(children, { "aria-describedby": `${uid}-annotations` })}
+      {React.cloneElement(children, { "aria-describedby": `${uid}-annotations` } as React.HTMLAttributes<HTMLElement>)}
```
@kaaloo tu as eu des soucis de build ?
Ces composants ne sont pas inclus sur l'app mobile du coup je me demande pourquoi ces modifs dans cette PR :)

kaaloo replied:
Yes des soucis dans le typage en relation avec la mise à jour de certaines dépendances malheureusement...

ledjay commented on apps/client/src/__tests__/auth/reinitialiser-mot-de-passe/__snapshots__/reinitialiser-mot-de-passe.test.tsx.snap:
```diff
@@ -130,7 +130,6 @@ exports[`auth/reinitialiser-mot-de-passe renders page if no email 1`] = `
           id="input-:r1:"
           name="email"
           type="email"
-          value=""
```
@kaaloo étrange que la value ait disparu de ce snapshot

ledjay commented on apps/mobile/tsconfig.json:
```diff
@@ -5,8 +5,9 @@
     "allowJs": false,
     "module": "esnext",
-    "moduleResolution": "node",
+    "moduleResolution": "bundler",
```
@kaaloo je vais tout re-tester en profondeur sur iOS du coup :)
Ca peut avoir des implications profondes sur de vielles dépendances j'imagine ^^'

ledjay reviewed (approved):
@kaaloo LGTM !
Je m'interroge juste de voir des fichiers /ui et /client dans la PR.
Des soucis de build ?
Reste à tester en profondeur ;)

ledjay committed:
:bug: Fix translation loader

ledjay committed:
Merge branch 'dev' into luis/ri-904-mise-a-jour-a-expo-53

ledjay committed:
:bug: Fix selector error

ledjay committed:
:bug: Fix axios error

ledjay committed:
feat: update city search modal padding issues

ledjay committed:
:bug: Fix input quicks (lack of debounce + weird resets)

ledjay committed:
:recycle: Code review

kaaloo committed:
Merge pull request #3206 from refugies-info/jeremie/fix-ios-quirks

Jeremie/fix ios quirks

kaaloo committed:
Merge branch 'dev' into luis/ri-904-mise-a-jour-a-expo-53

kaaloo committed:
chore(deps): bump @types/react and @types/react-dom to v19

- Updated @types/react from 18.3.12 to 19.0.14
- Updated @types/react-dom from 18.3.5 to 19.2.1
- Removed peerDependencies constraint on react version from ui package
- Updated all Radix UI component type references to use new React 19 types

kaaloo committed:
```
fix(types): allow null in RefObject type parameters

- Updated triggerRef type in BookmarkedModal and ReactionModal to RefObject<HTMLButtonElement | null>
- Updated triggerRef type in Modal component to RefObject<HTMLElement | null>
- Updated thumbUpRef type in VoteLayoutStandard to RefObject<ThumbUpAnimatedRef | null>
```

kaaloo committed:
Merge branch 'dev' into luis/ri-904-mise-a-jour-a-expo-53

kaaloo commented:
@ledjay merging this since it's been around for a while and we seem to have gotten the kinks out.  There's an expo sdk 54 coming up soon... 😅
